seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -344,19 +385,31 @@ def test_validate_anvil_vcf(self, mock_file_iter, mock_file_exist, mock_utils_lo
         self.assertEqual(response.json()['error'],
                          'Invalid VCF file format - file path must end with .vcf or .vcf.gz or .vcf.bgz')
 
+        # test header errors
         mock_file_exist.return_value = True
-        mock_file_iter.return_value = ['##fileformat=VCFv4.2\n',
-                                       '#CHROM	POS	ID	REF	ALT	QUAL']  # incomplete header line
+        mock_file_iter.return_value = BASIC_META + BAD_INFO_META + BAD_FORMAT_META + BAD_HEADER_LINE + DATA_LINES
         response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_GZ_DATA_PATH))
         self.assertEqual(response.status_code, 400)
-        self.assertEqual(response.json()['error'],
-                         'No samples found in the provided VCF. This may be due to a malformed file')
```

Comment:
we still need to test the case where there is a valid VCF header but no samples

Reply:
In the original code, `no samples` is part of error handling which means that bad VCF could return no samples. The new code raises exceptions if there are any errors. How about I change the `validate_anvil_vcf` API function not to detect `no samples`? We can do that because we will have a missing sample checking in the following step.

Reply:
sure but it still needs to be an explicit "no samples" error and not a "missing header column" error